src/components/keyboard/KavButton.tsx

```diff
@@ -0,0 +1,86 @@
+import * as React from 'react'
+import { Keyboard, Platform } from 'react-native'
+import { KeyboardAccessoryView } from 'react-native-keyboard-accessory'
+import { cacheStyles } from 'react-native-patina'
+import { useSafeAreaInsets } from 'react-native-safe-area-context'
+
+import { useSelector } from '../../types/reactRedux'
+import { EdgeButton } from '../buttons/EdgeButton'
+import { Theme, useTheme } from '../services/ThemeContext'
+import { MAX_TAB_BAR_HEIGHT } from '../themed/MenuTabs'
+
+interface KavButtonProps {
+  children?: React.ReactNode
+  disabled?: boolean
+  hasNotifications?: boolean
+  hasTabs?: boolean
+  label?: string
+  spinner?: boolean
+  testID?: string
+  visible?: boolean
+  onPress?: () => void | Promise<void>
+}
+
+/**
+ * A keyboard accessory button that spans most of the width of the screen,
+ * positioned above the keyboard.
+ *
+ * IMPORTANT: This component MUST be placed at the ROOT level of your component
+ * hierarchy (as a direct child of SceneWrapper) for proper keyboard positioning.
+ * TODO: Consider moving this to SceneWrapper since there is a lot of duplicate
+ * inset logic
+ */
+export const KavButton = (props: KavButtonProps): React.ReactNode => {
+  const { children, disabled = false, label, onPress, spinner = false, visible = true, testID, hasTabs = false, hasNotifications = false } = props
+  const theme = useTheme()
+  const styles = getStyles(theme)
+  const safeAreaInsets = useSafeAreaInsets()
+  const isIos = Platform.OS === 'ios'
+
+  const notificationHeight = useSelector(state => state.ui.notificationHeight)
+
+  const [isKeyboardVisible, setKeyboardVisible] = React.useState(false)
+
+  React.useEffect(() => {
+    // Listening event is different between iOS and Android, but actually mean
+    // the same thing: when the keyboard begins its movement
+    const keyboardDidShowListener = Keyboard.addListener(isIos ? 'keyboardWillShow' : 'keyboardDidShow', () => {
+      setKeyboardVisible(true)
+    })
+    const keyboardDidHideListener = Keyboard.addListener(isIos ? 'keyboardWillHide' : 'keyboardDidHide', () => {
+      setKeyboardVisible(false)
+    })
+
+    return () => {
+      keyboardDidShowListener.remove()
+      keyboardDidHideListener.remove()
+    }
+  }, [isIos])
+
+  const maybeTabBarHeight = hasTabs ? MAX_TAB_BAR_HEIGHT : 0
+  const maybeNotificationHeight = hasNotifications ? notificationHeight : 0
+
+  const keyboardAccessoryStyle = React.useMemo(
+    () => ({
+      ...styles.keyboardAccessory,
```

Comment:
Maybe get rid of the `cacheStyles`, and just put `backgroundColor: 'transparent'` directly in here. We're going to burn way more CPU cycles on the caching than we are on this one style property.